### ISSUE TITLE
Prompt users to keep annotation extensions when editing phenotype term

### ISSUE THREAD

**Issue by jseager7** [pombase/canto#1963]
(Requested by @ValWood)

In the previous meeting we discussed potential problems with the fact that the 'Copy and edit' option on phenotype annotations automatically copies the annotation extensions as well.

Rather than defaulting to _not_ copying the extensions, we thought it would be sensible to prompt the users whether they want to keep their annotation extensions whenever the user changes a phenotype term.

This prompt could be a simple yes/no choice, meaning users either keep or lose all of their extensions; or it could give users the option to select some extensions to keep.

@ValWood would you still want the users to be prompted when all the existing extensions apply to the new term, or should they only be prompted when the new term has a different set of extensions from the old term?

(For example, changing from a term _A_ that is in the range of extension _X_ to term _B_ that is also in the range of extension _X_ won't prompt, whereas changing _A_ to a term _C_ that is in the range of extension _X_ and _Z_ will prompt.)

@kimrutherford One thing I'm not sure about is whether it's possible to edit a phenotype term such that the new term isn't in the range of the extension. Is the extension automatically removed in this case? Does Canto warn if it will be removed?

**Comment by ValWood** [comment]
I think the prompt should apply even if the range is the same.

This should be a configurable option when it is added, and maybe should not exist in admin mode, because the prompt might be annoying for "super users"

We also need to think about if users should also  be prompted to add an extension if the term they are copying did not have one.

**Comment by ValWood** [comment]
I put this as lower   priority for now  - would like  to discuss further with PomBAse curators

**Comment by kimrutherford** [comment]
> Is the extension automatically removed in this case?

Not at the moment.  That needs to be fixed.

**Comment by kimrutherford** [comment]
> Does Canto warn if it will be removed?

I forgot to say: there's no warning either.

**Comment by ValWood** [comment]
![mockup-copy-edit-pop-up](https://user-images.githubusercontent.com/7359272/70713156-4ccbdb80-1cdd-11ea-9577-d4581a728810.jpg)

**Comment by ValWood** [comment]
We had a chat about this and curators aren't keen on having to actively say they want to keep an extension (although we did already suggest that this option would not be for admins, hidden comment above). The pop up is currently a bit busy overall and the "annotation extension" option somehow is not as visible as the others.

Here are a number of small changes to the layout that we think will make the annotation extension curation more obvious (And improve the look more generally).

 - [ ] 1. No need to show curator here?

 - [ ] 2. Make the term suggestion and definition boxes much smaller ( definition box can still be expandable)

 - [ ] 3. We were going to shorten the text for conditions, but it might not even be required at this point since  annotations have already been made? As a reminder, this could be available as pop up help when hovering over the menu item "conditions".  (Check, do we have a condition pop-up help text elsewhere?)

 - [ ] 4. I'm not even sure we need to say "Previously used conditions, (click to add to list)? If we still need something "click to add" would be sufficient? Presumably, the user will recognise that these are conditions they added previously?

 - [ ] 5. Other ' basic' help could be added in the menu

 - [ ] 6. If an annotation extension exists, display it with an "x" delete icon so it is easy to remove without entering the extension interface

 - [ ] 7. Change the "Edit" button to "Add". This will also make more sense if there are no existing extensions

 - [ ] 8.  Move the "Add another" link to the r.h.s. of the picker 

- [ ]  9. Make sure there is a bit of white space (consistent)  between each annotation type (vertically)

Once the clean up is done, we can think about whether it would be useful to split up and have a box for each available extension type ( I would quite like to do this)
so for example a kinase would have a "has substrate" box (even though we would still need to click through to the extension interface). This would alert the user about the range of available extensions for a term.

@Antonialock @mah11  I added  a couple more things from our discussion, so please check that these all sound OK

**Comment by jseager7** [comment]
> We were going to shorten the text for conditions, but it might not even be required at this point since annotations have already been made?

I'm concerned if we remove this text that curators could mistakenly think that the previously used conditions are conditions that have already been added to the annotation (basically mixing up the conditions picker with the previously used conditions). We could make the text a bit more discreet and reword it to something like 'Click to add previously used conditions'.

> Other ' basic' help could be added in the menu

Which menu is this?

> Move the "Add another" link to the r.h.s. of the picker

I think that might be a PomBase-specific feature, because I can't see it on my PHI-Canto versions. I'll check this later.

> Change the "Edit" button to "Add". This will also make more sense if there are no existing extensions

Maybe a better option is having individual Edit buttons for each extension, and an Add button that's always visible.

**Comment by mah11** [comment]
> we can think about whether it would be useful to split up and have a box for each available extension type ( I would quite like to do this)

Really? Even for modifications?

![Screenshot_2019-12-12 PomBase community curation tool - PMID 27935167 summary](https://user-images.githubusercontent.com/5196259/70733371-19e80e80-1d02-11ea-85f3-244179e61db8.png)

**Comment by ValWood** [comment]
Possibly not. Just throwing it in there as a suggestion.
